extensions/src/main/java/org/pentaho/platform/osgi/KarafBoot.java

```diff
@@ -196,7 +196,13 @@
               return false;
             } else if ( symlinks.contains( relativePath ) ) {
               File linkFile = new File( destDir, relativePath );
-              linkFile.getParentFile().mkdirs();
+              try {
+                boolean linkFileDirCreated = linkFile.getParentFile().mkdirs();
+                logger.info(
+                  "link file " + linkFile.getAbsolutePath() + " parent directory " + ( linkFileDirCreated ? "created" : "already existed" ) );
+              } catch ( SecurityException exception ) {
+                logger.warn( linkFile.getAbsolutePath() + " cannot create directory. Access denied." );
```

Comment:
If we can't create the directory, what should happen? Should execution be allowed to proceed as you have it here?
Is it better to stop execution and throw an exception because we don't know how to handle it?